agent/consul/state/intention_test.go

```diff
@@ -1083,6 +1085,124 @@ func TestStore_LegacyIntention_Snapshot_Restore(t *testing.T) {
 	}()
 }
 
+func TestStore_IntentionDecision(t *testing.T) {
```

Comment:
Might want to make a comment here about why this test isn't split into legacy/config-entry flavors like all of the other intention tests here.

Reply:
Added a comment